install-pdm.py fails if there's an invalid pyproject.toml in the current directory

- [x] I have searched the issue tracker and believe that this is not a duplicate.

**Make sure you run commands with `-v` flag before pasting the output.**

## Steps to reproduce

- Make an invalid `pyproject.toml` file in the current directory
- Run the installer: `curl -sSL https://pdm-project.org/install-pdm.py | python3 -` as specified [here](https://pdm-project.org/latest/#recommended-installation-method)

## Actual behavior

The installer errors out:

```sh
Run curl -sSL https://pdm-project.org/install-pdm.py | python3 -
Traceback (most recent call last):
  File "/home/runner/.local/bin/pdm", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/runner/.local/share/pdm/venv/lib/python3.12/site-packages/pdm/core.py", line 346, in main
    return core.main(args or sys.argv[1:])
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/.local/share/pdm/venv/lib/python3.12/site-packages/pdm/core.py", line 237, in main
    args = self._get_cli_args(args, obj)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/.local/share/pdm/venv/lib/python3.12/site-packages/pdm/core.py", line 220, in _get_cli_args
    config = project.pyproject.settings.get("options", {})
             ^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.4/x64/lib/python3.12/functools.py", line 995, in __get__
    val = self.func(instance)
          ^^^^^^^^^^^^^^^^^^^
  File "/home/runner/.local/share/pdm/venv/lib/python3.12/site-packages/pdm/project/core.py", line 125, in pyproject
    return PyProject(self.root / self.PYPROJECT_FILENAME, ui=self.core.ui)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/.local/share/pdm/venv/lib/python3.12/site-packages/pdm/project/toml_file.py", line 18, in __init__
    self._data = self.read()
                 ^^^^^^^^^^^
  File "/home/runner/.local/share/pdm/venv/lib/python3.12/site-packages/pdm/project/project_file.py", line 27, in read
    data = super().read()
           ^^^^^^^^^^^^^^
  File "/home/runner/.local/share/pdm/venv/lib/python3.12/site-packages/pdm/project/toml_file.py", line 23, in read
    return super().read()
           ^^^^^^^^^^^^^^
  File "/home/runner/.local/share/pdm/venv/lib/python3.12/site-packages/tomlkit/toml_file.py", line 45, in read
    return loads(content)
           ^^^^^^^^^^^^^^
  File "/home/runner/.local/share/pdm/venv/lib/python3.12/site-packages/tomlkit/api.py", line 46, in loads
    return parse(string)
           ^^^^^^^^^^^^^
  File "/home/runner/.local/share/pdm/venv/lib/python3.12/site-packages/tomlkit/api.py", line 86, in parse
    return Parser(string).parse()
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/.local/share/pdm/venv/lib/python3.12/site-packages/tomlkit/parser.py", line 139, in parse
    item = self._parse_item()
           ^^^^^^^^^^^^^^^^^^
  File "/home/runner/.local/share/pdm/venv/lib/python3.12/site-packages/tomlkit/parser.py", line 238, in _parse_item
    return self._parse_key_value(True)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/.local/share/pdm/venv/lib/python3.12/site-packages/tomlkit/parser.py", line 311, in _parse_key_value
    key = self._parse_key()
          ^^^^^^^^^^^^^^^^^
  File "/home/runner/.local/share/pdm/venv/lib/python3.12/site-packages/tomlkit/parser.py", line 360, in _parse_key
    return self._parse_bare_key()
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/.local/share/pdm/venv/lib/python3.12/site-packages/tomlkit/parser.py", line 404, in _parse_bare_key
    raise self.parse_error(EmptyKeyError)
tomlkit.exceptions.EmptyKeyError: Empty key at line 1 col 0
Traceback (most recent call last):
  File "<stdin>", line 463, in <module>
  File "<stdin>", line 459, in main
  File "<stdin>", line 376, in install
  File "<stdin>", line 346, in _post_install
  File "/opt/hostedtoolcache/Python/3.12.4/x64/lib/python3.12/subprocess.py", line 413, in check_call
Installing PDM (2.17.3): Creating virtual environment
Installing PDM (2.17.3): Installing PDM and dependencies
Installing PDM (2.17.3): Making binary at /home/runner/.local/bin
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/home/runner/.local/bin/pdm', '--help']' returned non-zero exit status 1.
Error: Process completed with exit code 1.
```

## Expected behavior

The installer should succeed regardless of what's in the current directory. 

## Environment Information

- OS: Linux, Ubuntu-latest
- python: 3.12

```bash
# Paste the output of `pdm info && pdm info --env` below:
# Not applicable since the issue is about the installer
```

## Demo

I've created a repo to demonstrate this error with Github Actions: <https://github.com/PowerSnail/pdm-install-error-sample>.

The CI starts a clean new ubuntu environment, installs python 3.12, and attempts to run `curl -sSL https://pdm-project.org/install-pdm.py | python3 -`.

Here's the error page: <https://github.com/PowerSnail/pdm-install-error-sample/actions/runs/10288375454/job/28473644150>

--- a/news/3085.bugfix.md
+++ b/news/3085.bugfix.md
@@ -1,0 +1,1 @@
+Don't fail `install-pdm.py` if there is an invalid `pyproject.toml` file under the current directory.

--- a/src/pdm/core.py
+++ b/src/pdm/core.py
@@ -21,6 +21,8 @@ from pathlib import Path
 from tempfile import TemporaryDirectory
 from typing import TYPE_CHECKING, cast
 
+import tomlkit
+import tomlkit.exceptions
 from resolvelib import Resolver
 
 from pdm import termui
@@ -217,7 +219,11 @@ class Core:
         project = self.create_project(is_global=False) if obj is None else obj
         if project.is_global:
             return args
-        config = project.pyproject.settings.get("options", {})
+        try:
+            config = project.pyproject.settings.get("options", {})
+        except tomlkit.exceptions.TOMLKitError as e:
+            self.ui.error(f"Failed to parse pyproject.toml: {e}")
+            config = {}
         (pos, command) = self.get_command(args)
         if command and command in config:
             # add args after the command

--- a/src/pdm/core.py
+++ b/src/pdm/core.py
@@ -221,7 +221,7 @@ class Core:
             return args
         try:
             config = project.pyproject.settings.get("options", {})
-        except tomlkit.exceptions.TOMLKitError as e:
+        except tomlkit.exceptions.TOMLKitError as e:  # pragma: no cover
             self.ui.error(f"Failed to parse pyproject.toml: {e}")
             config = {}
         (pos, command) = self.get_command(args)

--- a/src/pdm/core.py
+++ b/src/pdm/core.py
@@ -21,7 +21,6 @@ from pathlib import Path
 from tempfile import TemporaryDirectory
 from typing import TYPE_CHECKING, cast
 
-import tomlkit
 import tomlkit.exceptions
 from resolvelib import Resolver
 

